### PR TITLE
feat: save title or description of insight or dashboard on blur

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

see #11552 and #11551

## Changes

* saves insight/description name or description on blur
* saves dashboard name or description on blur
* allows onMouseDown handler for LemonButton (or else cancel click and input blur fight each other)

### insight interactions

insight-undo-v2

### dashboard interactions

dashboard-undo

## How did you test this code?

running it locally

### CODE REVIEW

**Review by clarkus — APPROVED** [review]

File: frontend/src/scenes/insights/insightLogic.ts [hunk 1]
```diff
@@ -236,8 +241,30 @@ export const insightLogic = kea<insightLogicType>({
...
+
                     savedInsightsLogic.findMounted()?.actions.loadInsights()
                     dashboardsModel.actions.updateDashboardInsight(updatedInsight)
+
+                    lemonToast.success(`Updated insight`, {
+                        button: {
+                            label: 'Undo',
+                            action: async () => {
+                                const response = await api.update(
+                                    `api/projects/${teamLogic.values.currentTeamId}/insights/${values.insight.id}`,
```
Comment (mariusandra) [inline frontend/src/scenes/insights/insightLogic.ts:253]: Can we do `api.get.the.url.somehow()`?

File: frontend/src/scenes/insights/insightLogic.ts [hunk 2]
```diff
@@ -236,8 +241,30 @@ export const insightLogic = kea<insightLogicType>({
...
+                                )
+                                // only update the fields that we changed
+                                const revertedInsight = { ...values.insight } as InsightModel
+                                for (const key of Object.keys(beforeUpdates)) {
+                                    revertedInsight[key] = response[key]
+                                }
+                                savedInsightsLogic.findMounted()?.actions.loadInsights()
+                                dashboardsModel.actions.updateDashboardInsight(revertedInsight)
+                                actions.setInsight(revertedInsight, { overrideFilter: false, fromPersistentApi: true })
+                                lemonToast.success('Insight change reverted')
```
Comment (mariusandra) [inline frontend/src/scenes/insights/insightLogic.ts:264]: Will you consider me a bastard if I think this should be tested 😅

File: frontend/src/lib/components/EditableField/EditableField.tsx [hunk 3]
```diff
@@ -63,6 +65,12 @@ export function EditableField({
 
     const isSaveable = !minLength || tentativeValue.length >= minLength
 
+    const mouseDownOnCancelButton = (e: React.MouseEvent): void => {
+        // if saveOnBlur is set the onBlur handler of the input fires before the onClick event of the button
+        // this onMouseDown handler fires before the input can see the click and fire onBlur
+        e.preventDefault()
+    }
```
Comment (mariusandra) [inline frontend/src/lib/components/EditableField/EditableField.tsx:72]: Calling `preventDefault` on a `mouseDown` event on a button prevents the `onBlur` on a loosely related input from running...? 🤔 

Seems like sorcery, but it works when I tested it, so 🍨

**Review by mariusandra — APPROVED** [review]
Left some inline thoughts, and also think we can probably "cancel" instead of saving if the user didn't change any text (prevents a flicker), but 👍

File: frontend/src/scenes/insights/insightLogic.ts (hunk 2 above)
Comment (pauldambra) [reply frontend/src/scenes/insights/insightLogic.ts:264]: I don't see how a bug could sneak in there 🤣

File: frontend/src/scenes/insights/insightLogic.ts (hunk 1 above)
Comment (pauldambra) [reply frontend/src/scenes/insights/insightLogic.ts:253]: From laziness and focus perspective. All of the URLs in this logic are boutique and locally-sourced. Let's tidy them in one go :)

File: frontend/src/scenes/insights/insightLogic.ts (hunk 2 above)
Comment (pauldambra) [reply frontend/src/scenes/insights/insightLogic.ts:264]: Hmmm, testing this isn't straightforward. I think this will need to be Cypress

**Pull request merged**